### DISCUSSION TITLE
What about EDC and DSC compatibility?

### DISCUSSION THREAD

**Discussion by git-masoud** [eclipse-edc/Connector#1549] [Q&A] [answered]
Hello guys,
Looking at the fact both EDC and DSC use IDS data protocol, we were under the impression that we could easily swap one for the other without too much hassle.
However, we’ve noticed that there are some differences in the format of the exchanged messages,
PR https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/1541 seems to focus exactly on that, bringing JSON-LD support to EDC. Furthermore, I had a quick chat with @juliapampus , and she also confirmed that currently EDC and DSC are not able to talk to each other. 
Question is, do we have an exhaustive list of compatibility issues between these two connectors? In other words, once the above is resolved, do we expect it to work immediately, or there are some more things to consider.

Thanks!

**Comment by jimmarino** [comment, marked as answer]
Wire compatibility with the DSC is currently not on the roadmap and won't likely be something that is undertaken by the core development community. 

We will be producing specifications for EDC's messaging flows and protocols, which could be used by a group that wants to make DSC compatible with the EDC.

**Reply by mspiekermann** [reply]
> We will be producing specifications for EDC's messaging flows and protocols, which could be used by a group that wants to make DSC compatible with the EDC.

Worth to mention, that this is done in close alignment with IDSA and together with a group of people responsible for providing coresponding components (DSC among others). Within this group we will further work on a technical specification, which right now has some identified gaps. It will specify the concepts of already published IDS RAM 4.0 and asap allow components to be wired as expected. When this is in place, EDC will support this protocol and thereby all other IDSA compliant connectors.
